geode-core/src/main/java/org/apache/geode/internal/cache/InitialImageOperation.java

```diff
@@ -846,12 +846,12 @@ boolean processChunk(List entries, InternalDistributedMember sender)
       }
       List<Entry> entriesToSynchronize = new ArrayList<>();
 
+      if (internalDuringApplyDelta != null && !internalDuringApplyDelta.isRunning
```

Comment:
It looks to me like changing this test hook, by moving it out of the loop, will break the existing TombstoneDUnitTest that uses DuringApplyDelta. Why did you need to change it?

Reply:
because I need to make sure processChunk is called. If I did not move out of the loop, if the entryCount is 0, the testHook will not be called. I don't want to introduce another testHook. 

I care if processChunk is call more than if there's any data in chunk.